botocore/auth.py

```diff
@@ -303,14 +304,17 @@ def signature(self, string_to_sign, request):
         key = self.credentials.secret_key
         k_date = self._sign(('AWS4' + key).encode('utf-8'),
                             request.context['timestamp'][0:8])
-        k_region = self._sign(k_date, self._region_name)
+        k_region = self._sign(k_date, self._signing_region)
         k_service = self._sign(k_region, self._service_name)
         k_signing = self._sign(k_service, 'aws4_request')
         return self._sign(k_signing, string_to_sign, hex=True)
 
     def add_auth(self, request):
         if self.credentials is None:
             raise NoCredentialsError
+        signing_context = request.context.get('signing', {})
```

Comment:
Just want to make sure here. Each request gets one auth instance and individual requests can never share auth instances? If that is so lets make a note here as if we ever do consider sharing them that could be a hard concurrency issue to figure out.